lib/web_ui/flutter_js/src/canvaskit_loader.js

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import { engineRevision } from "./revision.js";
+import { createWasmInstantiator } from "./instantiate_wasm.js";
+
+export const loadCanvasKit = (deps, config, browserEnvironment) => {
+  window.flutterCanvasKitLoaded = new Promise((resolve, reject) => {
+    const supportsChromiumCanvasKit = browserEnvironment.hasChromiumBreakIterators && browserEnvironment.hasImageCodecs;
+    if (!supportsChromiumCanvasKit && config.canvasKitVariant == "chromium") {
+      throw "Chromium CanvasKit variant specifically requested, but unsupported in this browser";
+    }
+    const useChromiumCanvasKit = supportsChromiumCanvasKit && (config.canvasKitVariant !== "full");
+    let baseUrl = config.canvasKitBaseUrl ?? `https://www.gstatic.com/flutter-canvaskit/${engineRevision}/`;
+    if (useChromiumCanvasKit) {
+      baseUrl = `${baseUrl}/chromium/`;
+    }
+    let canvasKitUrl = `${baseUrl}canvaskit.js`;
+    if (deps.flutterTT.policy) {
+      canvasKitUrl = deps.flutterTT.policy.createScriptURL(canvasKitUrl);
+    }
+    const wasmInstantiator = createWasmInstantiator(`${baseUrl}canvaskit.wasm`);
```

Comment:
(Unless `''` is a valid `baseUrl`, I guess)

Reply:
I originally had it the way you're suggesting, because largely I agree with you, but that's not how our code actually works right now. The `canvasKitBaseUrl` is expected to have a trailing slash currently. We need to respect the current semantics.